### DISCUSSION TITLE
After updating to version 2.3.4 on Home Assistant, the Denon AirPlay player provider disappeared

### DISCUSSION THREAD

**Discussion by garisroot** [music-assistant/support#3346] [Q&A] [answered]
Hi,

After the recent update  (2.3.4) to the Music Assistant add-on in Home Assistant, the Denon player provider became unavailable. I tried reloading the integration a few times, but it didn’t resolve the issue. I then removed the Music Assistant integration and reinstalled it, but the Denon player provider still didn’t appear in the list.

AirPlay works fine with Denon on other devices, so there doesn’t seem to be an issue with the device itself.

Does this mean that Denon is no longer supported after the update, or am I missing something? Thanks for your help!

**Comment by OzGav** [comment]
Try the HEOS integration

**Reply by garisroot** [reply]
But in the list HEOS player still is not available. I have separate HEOS integration, but entities are not available on Music Assistant. 
<img width="332" alt="Screenshot 2024-12-22 at 9 41 37 PM" src="https://github.com/user-attachments/assets/977d1918-93e3-41a2-959a-6ab45556ca93" />

**Reply by OzGav** [reply, marked as answer]
You need to add those through the Home Assistant MediaPlayer provider

![IMG_0063](https://github.com/user-attachments/assets/189e9a28-6641-4981-bf9b-48a29ae1ba23)

**Reply by garisroot** [reply]
Thanks! Issue is solved :)
